build/gutenberg-block.js

```diff
@@ -1 +1,154 @@
+                        return "fab fa-foursquare";
+                    } else if (parsedUrl.hostname.includes("instagram.com")) {
+                        return "fab fa-instagram";
+                    } else if (parsedUrl.hostname.includes("pinterest.com")) {
+                        return "fab fa-pinterest";
+                    } else if (parsedUrl.hostname.includes("linkedin.com")) {
+                        return "fab fa-linkedin";
+                    } else {
+                        return null;
+                    }
+                } catch (_) {
+                    return null;
+                }
+            };
+
+            const socialIcon = getSocialIcon(i);
+
+            return (
+                e.createElement("div", null,
+                    e.createElement(l, {
+                        label: "Select which field you want to show",
+                        value: i,
+                        options: r.map(e => ({ label: e.label, value: e.value })),
+                        onChange: e => { o({ option: e }) }
+                    }),
+                    e.createElement("div", null,
+                        isImageUrl(i) ?
+                            e.createElement("p", null,
+                                e.createElement("span", { className: "css-1imalal" }, "Preview:"),
+                                e.createElement("br", null),
+                                e.createElement("img", { src: i.replace(/,+$/, ''), alt: "Business Profile Image", style: { width: "100px", height: "100px" } })
+                            ) :
+                        socialIcon ?
+                            e.createElement("p", null,
+                                e.createElement("span", { className: "css-1imalal" }, "Preview:"),
+                                e.createElement("br", null),
+                                e.createElement("a", { href: i, target: "_blank", rel: "noopener noreferrer" },
+                                    e.createElement("i", { className: socialIcon })
+                                )
+                            ) :
+                            e.createElement("p", null, i)
+                    )
+                )
+            );
+        },
+        save: ({ attributes: t }) => {
+            const { option: l } = t;
+
+            const isImageUrl = (url) => {
+                try {
+                    // Remove trailing commas before parsing the URL
+                    const cleanUrl = url.replace(/,+$/, '');
+                    const parsedUrl = new URL(cleanUrl);
+                    return parsedUrl.hostname === 'media-prod.apigateway.co';
+                } catch (e) {
+                    return false;
+                }
+            };
+
+            const getSocialIcon = (url) => {
+                try {
+                    const parsedUrl = new URL(url);
+                    if (parsedUrl.hostname.includes("facebook.com")) {
```

Comment:
Can you use a switch case here as well?

Reply:
@Padminia Replaced the try-catch block with a switch-case statement for handling social links.